Arrangement_on_surface_2/include/CGAL/Arr_geometry_traits/Bezier_x_monotone_2.h

```diff
@@ -959,11 +959,12 @@ _Bezier_x_monotone_2<RatKer, AlgKer, NtTrt, BndTrt>::point_position
 
 
   if ( p.is_rational() ){
-    const Rational& px = ((Rat_point_2) p).x();
```

Comment:
Rat_point_2::x() returns a value, so capturing it by const ref is UB, and it indeed caused problems.

Reply:
Strange. Were you able to reproduce the crash and was this diff fixing the segfault?

Reply:
With some test cases, it would segfault, and in others it would get assertion error, I didn't check the cause of segfault before, but yes, now I checked it. The crash is reproduced consistently, and this line was the cause of the segfault.

Reply:
However, this change alone doesn't fix the bug.

Reply:
I checked out the reason of this behavior. It comes out to the fact that returning a `const ref` from a `pr value` is not safe. The `pr value` may be destructed, and with it its member variables.
Example: 
https://gist.github.com/theartful/d3a8b85e8d8e9cbe465bf570b740f8ab

Reply:
The problem here is that it is a const ref to a **sub-object** of a PR-value. If it was to the PR-value itself, then the lifetime extension would have applied. See https://en.cppreference.com/w/cpp/language/reference_initialization#Lifetime_of_a_temporary. But binding a reference to a sub-object does not extend the lifetime of the object.

Reply:
Nice! Thanks for clearing that up.